plugins/bulk-renamer/RenamerDialog.vala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2010-2017  Vartan Belavejian
+ * Copyright (C) 2019-2020 Jeremy Wootten
+ *
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  Authors:
+ *  Vartan Belavejian <https://github.com/VartanBelavejian>
+ *  Jeremy Wootten <jeremywootten@gmail.com>
+ *
+*/
+
+public class Files.RenamerDialog : Gtk.Dialog {
+    private Files.Renamer renamer;
+    private Gtk.Widget rename_button;
+    public RenamerDialog (string _basename, GLib.File[] _files) {
+        renamer.add_files (_files);
+        renamer.set_base_name (_basename);
+    }
+
+    construct {
+        deletable = true;
```

Comment:
Dialogs shouldn't have close window controls since it's ambiguous what happens when the window is closed. 
```suggestion
```